pkg/console/operator/status.go

```diff
@@ -48,56 +50,68 @@ import (
+//   c.handleDegraded(operatorConfig, "RouteStatus", "FailedHost", error.New("route is not available at canonical host..."))
+//   - Type RouteStatusDegraded
+//   - error string value is used as message
+func (c *consoleOperator) HandleDegraded(operatorConfig *operatorsv1.Console, typePrefix string, reason string, err error) {
+	// TODO: typePrefix & reason are not the same thing.
+	// TypePrefix is generic, may have many reasons. Don't concat reason.
+	conditionType := typePrefix + operatorsv1.OperatorStatusTypeDegraded
 	handleCondition(operatorConfig, conditionType, reason, err)
 }
 
-func (c *consoleOperator) HandleProgressing(operatorConfig *operatorsv1.Console, prefix string, err error) {
-	conditionType := prefix + operatorsv1.OperatorStatusTypeProgressing
-	handleCondition(operatorConfig, conditionType, prefix, err)
+func (c *consoleOperator) HandleProgressing(operatorConfig *operatorsv1.Console, typePrefix string, reason string, err error) {
+	// TODO: typePrefix & reason are not the same thing.
+	// TypePrefix is generic, may have many reasons. Don't concat reason.
+	conditionType := typePrefix + operatorsv1.OperatorStatusTypeProgressing
+	handleCondition(operatorConfig, conditionType, reason, err)
 }
 
-func (c *consoleOperator) HandleAvailable(operatorConfig *operatorsv1.Console, prefix string, err error) {
-	conditionType := prefix + operatorsv1.OperatorStatusTypeAvailable
-	handleCondition(operatorConfig, conditionType, prefix, err)
+func (c *consoleOperator) HandleAvailable(operatorConfig *operatorsv1.Console, typePrefix string, reason string, err error) {
+	// TODO: typePrefix & reason are not the same thing.
+	// TypePrefix is generic, may have many reasons. Don't concat reason.
+	conditionType := typePrefix + operatorsv1.OperatorStatusTypeAvailable
+	handleCondition(operatorConfig, conditionType, reason, err)
+}
+
+// TODO: does this helper need to be removed since it convolutes the error flip even more?
+// HandleProgressingOrDegraded exists until we remove type SyncError
+// If isSyncError
+// - Type suffix will be set to Progressing
+// else
+// - Type suffix will be set to Degraded
+//
+// TODO: what if it is nil?
+// - Will we wedge on Degraded never being flipped?
+// - Will it still do what its supposed to do?
+func (c *consoleOperator) HandleProgressingOrDegraded(operatorConfig *operatorsv1.Console, typePrefix string, reason string, err error) {
+	if errors.IsSyncError(err) {
+		c.HandleDegraded(operatorConfig, typePrefix, reason, nil)
+		c.HandleProgressing(operatorConfig, typePrefix, reason, nil)
+	} else {
+		c.HandleDegraded(operatorConfig, typePrefix, reason, nil)
+		c.HandleProgressing(operatorConfig, typePrefix, reason, nil)
+	}
 }
 
 // internal func for handling conditions
-func handleCondition(operatorConfig *operatorsv1.Console, conditionType string, reason string, err error) {
+func handleCondition(operatorConfig *operatorsv1.Console, conditionTypeWithSuffix string, reason string, err error) {
```

Comment:
shouldnt this be `conditionTypeWithPrefix` since we are prefixing the condition type with the action/funcName ?